ktor-features/ktor-jackson/jvm/test/io/ktor/tests/jackson/JacksonBlockingTest.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.jackson
+
+import io.ktor.application.*
+import io.ktor.client.request.*
+import io.ktor.features.*
+import io.ktor.http.*
+import io.ktor.jackson.*
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.server.testing.*
+import kotlinx.coroutines.*
+import java.io.*
+import java.lang.reflect.*
+import java.util.concurrent.*
+import kotlin.coroutines.*
+import kotlin.test.*
+
+class JacksonBlockingTest {
+    private val dispatcher = UnsafeDispatcher()
+    private val environment = createTestEnvironment {
+        parentCoroutineContext = dispatcher
+    }
+
+    @AfterTest
+    fun cleanup() {
+        dispatcher.close()
+    }
+
+    @Test
+    fun testReceive(): Unit = withApplication(environment) {
+        application.intercept(ApplicationCallPipeline.Setup) {
+            withContext(dispatcher) {
+                proceed()
+            }
+        }
+        application.install(ContentNegotiation) {
+            jackson()
+        }
+        application.routing {
+            post("/") {
+                assertEquals(K(77), call.receive())
+                call.respondText("OK")
+            }
+        }
+
+        runBlocking {
+            assertEquals("OK", client.post("/") {
+                body = "{\"i\": 77}"
+                contentType(ContentType.Application.Json)
+            })
+        }
+    }
+
+    data class K(var i: Int)
+
+    private class UnsafeDispatcher : CoroutineDispatcher(), Closeable {
```

Comment:
maybe extract this to a common module and reuse it?

Reply:
I would do this but I can't because of change restrictions. Will do in 1.5.0